### PR TITLE
feat: create initial version of ElasticsearchExporter

### LINKED ISSUE #21838: Create initial version of ElasticsearchExporter

### PR DESCRIPTION
**Pull request by mustafadagher**
## Description

create initial version of ElasticsearchExporter

## Related issues

closes #21838

### CODE REVIEW

File: zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/ElasticsearchExporter.java [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+    } catch (final PersistenceException ex) {
+      throw new ElasticsearchExporterException(ex.getMessage(), ex);
+    }
+  }
+
+  private void updateLastExportedPosition() {
+    controller.updateLastExportedRecordPosition(lastPosition, null);
+  }
+
+  private record ElasticsearchRecordFilter() implements RecordFilter {
```
Comment (mustafadagher) [inline zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/ElasticsearchExporter.java:142]: It's not obvious to me if the supported `RecordTypes` and `ValueTypes` will need to be configurable at some point

**Comment by mustafadagher** [comment]
The PR is missing some Integration tests for the exporter, could be added in a followup PR.

File: zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/clients/elasticsearch/ElasticsearchClientFactory.java [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+    } catch (final Exception e) {
+      throw new ElasticsearchExporterException("Couldn't connect to Elasticsearch. Abort.", e);
+    }
+  }
+
+  private HttpAsyncClientBuilder configureHttpClient(
+      final HttpAsyncClientBuilder httpAsyncClientBuilder,
+      final ElasticsearchProperties elsConfig) {
+    setupAuthentication(httpAsyncClientBuilder, elsConfig);
+    // TODO setup SSLContext if configured
```
Comment (mustafadagher) [inline zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/clients/elasticsearch/ElasticsearchClientFactory.java:97]: This is still missing.

**Comment by deepthidevaki** [comment]
I made the following changes to the PR

1. Removed the retry for health check when creating the ElasticsearchClient. The retries will block the Zeebe exporter thread for the whole time. It is enough to just check the health once and log the warning. We are not failing the client creation anyway. So there is no value in retrying for such a long time.
2. Renamed it to `CamundaExporter` to avoid confusion with the existing ElasticsearchExporter. My assumption is we will extend this to also support OpenSearch. If we end up having another implementation for Opensearch we can rename it then.
3. Added a single acceptance test to verify that everything works. More tests need to be added. 

I will add followup issues for the following:
Tests:
* Verify that basic authentication using configured username + password works
* Verify flush behavior
    * Should flush periodically
    * Should use flush configuration
* Verify position is updated after flushing
* Verify that record is exported when elastic is not initially reachable, but is reachable later.

Configuration:
* Configure bulksize if needed, and use it for flush
* Validate configuration

@EuroLew Could you do a second round of review?

File: zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/config/ElasticsearchExporterConfiguration.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  public String toString() {
+    return "ElasticsearchExporterConfiguration{"
+        + "bulk="
+        + bulk
+        + ", elasticsearch="
+        + elasticsearch
+        + '}';
+  }
+
+  public static class BulkConfiguration {
```
Comment (EuroLew) [inline zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/config/ElasticsearchExporterConfiguration.java:24]: If we're going to use getters and setters for config do it here as well?

**Review by EuroLew — APPROVED** [review]

**Pull request merged**